loki/transform/transform_hoist_variables.py

```diff
@@ -150,6 +151,8 @@ def transform_subroutine(self, routine, **kwargs):
         call_map = CaseInsensitiveDict((str(call.name), call) for call in calls)
 
         for child in successors:
+            if not isinstance(child, SubroutineItem):
```

Comment:
If SCC HOIST is used together with `GlobalVarOffloadTransformation`, some of the successors may be `Item`s. This loop only seems to look at `SubroutineItem`s, therefore adding this if statement.